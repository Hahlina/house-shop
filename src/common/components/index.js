export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
