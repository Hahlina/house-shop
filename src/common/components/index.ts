export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
