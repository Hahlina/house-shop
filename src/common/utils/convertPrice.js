export const convertPrice = (price) =>
    new Intl.NumberFormat('en-US', {
        style: 'currency',
        currency: 'USD'
    }).format(price || 0);
