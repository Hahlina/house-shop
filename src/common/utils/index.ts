export * from './convertPrice';
export * from './capitalizeFirstLetter';
