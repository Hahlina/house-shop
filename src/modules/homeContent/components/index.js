export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
