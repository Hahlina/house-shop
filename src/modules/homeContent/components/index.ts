export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
