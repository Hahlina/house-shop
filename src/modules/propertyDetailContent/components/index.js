export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
