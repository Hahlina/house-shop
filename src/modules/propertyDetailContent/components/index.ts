export * from './PropertyGallery/PropertyGallery';
export * from './PropertyGallery/PropertyGalleryLoader';
export * from './FullInformationCard/FullInformationCard';
export * from './FullInformationCard/FullInformationCardLoader';
export * from './PropertyDetailContentLoader/PropertyDetailContentLoader';
export * from './PropertyDetailBreadCrumbs/PropertyDetailBreadCrumbs';
