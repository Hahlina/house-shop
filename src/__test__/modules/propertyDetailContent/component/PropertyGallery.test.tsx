import { PropertyGallery } from 'modules/propertyDetailContent/components';

import { render, screen } from 'common/utils/test';

describe('PropertyGallery component', () => {
    const photos = ['photo1.jpg', 'photo2.jpg'];

    it('should render with photos without crashing', () => {
        render(<PropertyGallery photos={photos} />);
    });

    it('should contain two swiper-containers', () => {
        render(<PropertyGallery photos={photos} />);
        const containers = screen.getAllByTestId('swiper-container');
        expect(containers).toHaveLength(2);
    });

    it('should contain four slides', () => {
        render(<PropertyGallery photos={photos} />);
        const slides = screen.getAllByTestId('swiper-slide');
        expect(slides).toHaveLength(4);
    });

    it('should have correct width', () => {
        render(<PropertyGallery photos={photos} />);
        const containers = screen.getAllByTestId('swiper-container');
        containers.forEach((container) => {
            expect(container).toHaveStyle('width: 100%');
        });
    });

    it('should take a snapshot', () => {
        const { container } = render(<PropertyGallery photos={photos} />);
        expect(container).toMatchSnapshot();
    });
});
