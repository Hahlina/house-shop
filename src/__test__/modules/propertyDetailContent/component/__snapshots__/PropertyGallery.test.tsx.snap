// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyGallery component should take a snapshot 1`] = `
<div>
  <div
    class="sc-eDLKkx bSMyoG"
  >
    <swiper-container
      class="sc-jTQCzO lnbGVf"
      data-testid="swiper-container"
      init="false"
    >
      <swiper-slide
        data-testid="swiper-slide"
      >
        <img
          alt="property-0"
          src="photo1.jpg"
        />
      </swiper-slide>
      <swiper-slide
        data-testid="swiper-slide"
      >
        <img
          alt="property-1"
          src="photo2.jpg"
        />
      </swiper-slide>
    </swiper-container>
    <swiper-container
      class="sc-gLLuof EQlPQ"
      data-testid="swiper-container"
      init="false"
    >
      <swiper-slide
        data-testid="swiper-slide"
      >
        <img
          alt="property-thumbs-0"
          src="photo1.jpg"
        />
      </swiper-slide>
      <swiper-slide
        data-testid="swiper-slide"
      >
        <img
          alt="property-thumbs-1"
          src="photo2.jpg"
        />
      </swiper-slide>
    </swiper-container>
  </div>
</div>
`;
